### ISSUE TITLE
Generate ISOs headlessly so the distro installer can be automated from package updates.

### ISSUE THREAD

**Issue by kgilmer** [regolith-linux/regolith-desktop#486]

**Comment by atillabirer** [comment]
What is the obstacle in creating a Regolith 1.5 ISO? I have experience in the live-* scripts (live-build etc) and I could throw a hand of help if required. Amazing project by the way, have been using Regolith for a year now!

**Comment by kgilmer** [comment]
Hi @atillabirer , the current obstacle is in getting `live-build` to generate an Ubuntu live installer ISO that is able to:
1) boot on qemu, efi, and bios
2) launch and successfully complete the ubiquity installer process

In the past I had been using the Cubic GUI ISO builder, but am not really interested in continuing to use it for Regolith.  It's a great tool, just not the right one for a project like this.  We need something automatable and upgradable.

FWIW I've recently spent several days trying to get various branches, forks, patches, and so forth of live-build to produce a Ubuntu 20.04 installer without success.  I've had some partial successes here and there but nothing complete.  It's clearly possible as several other Ubuntu-derivatives rely on live-build to create their ISOs, but it's simply not something I've been able to get working.

What I need is to get a live-build config that generates just the stock Ubuntu environment, or even a reduced version.  From there I'm able to deal w/ the packaging configuration necessary to replace the Ubuntu desktop bits with Regolith.  Would love the help if you're able to make some progress @atillabirer !  FYI @moritzheiber has also been working on this, we have a private repo we can share if you're interested in collaborating.  But at this time as I mentioned the current stuff we have does not boot into the installer.

**Comment by atillabirer** [comment]
live-build is indeed f**** on Ubuntu. I have managed to build working Ubuntu ISOs by using a debian docker/vm/live boot using --mode ubuntu to build. You guys could give it a try as Debian people maintain their live-build well.

**Comment by kgilmer** [comment]
Interesting...do you have a link or gist or something I could try out @atillabirer ?

**Comment by atillabirer** [comment]
I will create an ISO today and send it to you, trying to include as many related things to Regolith as possible (regarding build and install).

Have you guys thought about rebasing to Debian? Ubuntu people seem to dislike things like live-build (they recently ditched their network installer mini.iso) and seem to be trying to force the Snap stuff.

**Comment by kgilmer** [comment]
Thanks, looking forward to it @atillabirer !  FWIW I only need the live-build configuration scripts, as I will generate an ISO myself for testing.  FWIW, when I looked into using Debian's live-build, I was able to generate a bootable ISO but was unable to get it to launch the ubiquity installer to completion.  Part of the issue was w/ the installer wanting to write to a read-only squashfs filesystem.  I was unable to crack that nut.

Regarding Debian, yes it's on the short list.  It was an early ask when we initially released.  In fact if an Ubuntu-based ISO cannot be done, we will likely offer Regolith on Ubuntu via PPA only and move to Debian for the ISO.
